tiledb/sm/rest/rest_client.cc

```diff
@@ -102,11 +102,13 @@ Status RestClient::get_array_schema_from_rest(
     const URI& uri, ArraySchema** array_schema) {
   // Init curl and form the URL
   Curl curlc;
-  RETURN_NOT_OK(curlc.init(config_, extra_headers_));
   std::string array_ns, array_uri;
   RETURN_NOT_OK(uri.get_rest_components(&array_ns, &array_uri));
-  std::string url = rest_server_ + "/v1/arrays/" + array_ns + "/" +
-                    curlc.url_escape(array_uri);
+  std::string cache_key = array_ns + ":" + array_uri;
+  RETURN_NOT_OK(curlc.init(
+      config_, extra_headers_, &cache_key, &redirect_meta_, &redirect_mtx_));
+  std::string url = redirect_uri(array_ns, array_uri) + "/v1/arrays/" +
```

Comment:
It looks like everywhere we invoke `redirect_uri`, we now have a `cache_key`. Modify `redirect_uri` to accept `cache_key` instead of `array_ns` and `array_uri`.